src/helm_manager.py

```diff
@@ -173,7 +217,16 @@ def run(self):
                 logging.error(f"Error during execution: {e}")
                 sys.exit(1)
         else:
-            logging.warning(f"Release {self.release_name} does not exist. You need to install it first.")
+            status = self._get_release_status()
+            print (status)
+            if status.get('STATUS') == 'pending-upgrade':
+                revision = str(int(status.get('REVISION'))-1)
```

Comment:
You could log something here, saying it'll do a rollback. I see there is a log after it did it.

Additionally on line 221 you have a status print, not sure if you need it still or was for debugging. Maybe a formatted log is preferred there?